vignettes/osat.Rmd

```diff
@@ -103,11 +103,10 @@ OSAT::get.experiment.setup(g_setup) |>
 # Compare scores with various implementations
 Compare OSAT score generated using designit.
 ```{r}
-OSAT::getLayout(gc) |>
+(OSAT::getLayout(gc) |>
   left_join(OSAT::get.experiment.setup(g_setup)) |>
   data.table::data.table() |>
-  osat_score("plates", c("SampleType", "Race", "AgeGrp")) |>
-  pull(score)
+  osat_score("plates", c("SampleType", "Race", "AgeGrp")))$score
```

Comment:
Mixing pipelines with `$` reduces readability.

We could avoid this by replacing `pull()` with `with()` from base R: it’s a drop-in replacement.

(But does it really make sense to drop ‘dplyr’ as a dependency?)

Reply:
`pull()` wouldn't work, since it only works for `data.frame()`s/`tibble()`s and we have a `list()` here. I also felt that `()$score` is unsatisfying, that's why I now use `pluck()` (1e9e5419fe1fb46c70cac6825bec817689009a5e). Is that better?

regarding `dplyr`, not sure I know what you meant. we are not dropping `dplyr` dependency, only `tidyselect` (by your suggestion).

btw, I'd love to drop `dplyr` dependency, mainly because it increases maintenance burden due to regular deprecations. but we are too dependent on it.

Reply:
I’d still slightly favour `with()` since it does’t require quoting the name, but either works and it’s a matter of personal preference.